SPECS/maven/maven.spec

```diff
@@ -0,0 +1,301 @@
+# @TODO
+# Since bootstrap has been removed for maven, it requires a pre-built maven binary to build itself.
+# Relying on 1.0 maven rpm to provide the mvn binary for the build.
+%ifarch x86_64
+Source1:        https://packages.microsoft.com/cbl-mariner/1.0/prod/base/x86_64/rpms/%{name}-%{mvn_1_0_pmc_ver}.cm1.x86_64.rpm#/%{name}-%{mvn_1_0_pmc_ver}.cm1.x86_64.rpm
+%endif
+%ifarch aarch64
+Source1:        https://packages.microsoft.com/cbl-mariner/1.0/prod/base/aarch64/rpms/%{name}-%{mvn_1_0_pmc_ver}.cm1.aarch64.rpm#/%{name}-%{mvn_1_0_pmc_ver}.cm1.aarch64.rpm
+%endif
+
+%if ! 0%{?sources_generation}
+%define offline_build -o
+%endif
+BuildRequires:  javapackages-local-bootstrap
+BuildRequires:  msopenjdk-11
+BuildRequires:  wget
+BuildRequires:  which
+Requires:       %{_bindir}/which
+Requires:       msopenjdk-11
+# In order to re-generate these sources after a version update, switch "sources_generation" to 1
+# and make sure network is enabled during the build. The tarballs will be inside the built 'maven-cached-sources' subpackage.
+%if ! 0%{?sources_generation}
+Source2:        %{m2_cache_tarball_name}
+Source3:        %{licenses_tarball_name}
+%endif
+
+%description
+Maven is a software project management and comprehension tool. Based on the concept of a project object model (POM). Maven can manage a project's build, reporting and documentation from a central piece of information.
+
+%if 0%{?sources_generation}
+%package cached-sources
+Summary:        NOT TO BE USED AS REGULAR PACKAGE! REQUIRES NETWORK ACCESS!
+
+%description cached-sources
+%{summary}
+This is an artificial package to ease generation of cached sources for
+the regular builds of "maven" when the "sources_generation" macro is set to 0.
+
+%endif
+
+%prep
+# Installing 1.0 maven rpm from PMC to provide prebuilt mvn binary.
+rpm -i --nodeps %{SOURCE1}
+which mvn
+
+%if ! 0%{?sources_generation}
+# Setup maven .m2 cache directory
+mkdir /root/.m2
+pushd /root/.m2
+tar xf %{SOURCE2} --no-same-owner
+popd
+%endif
+
+%setup -q -n apache-%{name}-%{version}
+%if ! 0%{?sources_generation}
+# Setup licenses. Remove LICENSE.vm script, which downloads all subproject license files, and replace with prepopulated license tarball.
+rm -v apache-maven/src/main/appended-resources/META-INF/LICENSE.vm
+pushd apache-maven
+tar xf %{SOURCE3} --no-same-owner
+cp -v ./target/licenses/lib/* %{_var}/opt/apache-maven/lib
+popd
+%endif
+
+%build
+# Changing distribution dir to BUILD directory as install macro clears buildroot prior to creating a fresh directory, thus clearing artifacts copied by maven. We copy them later.
+MAVEN_DIST_DIR=%{_builddir}%{_prefixmvn}
+
+export JAVA_HOME="%{_libdir}/jvm/msopenjdk-11"
+export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$(find $JAVA_HOME/lib -name "jli")
+
+sed -i 's/www.opensource/opensource/g' DEPENDENCIES
+pwd
+echo $LD_LIBRARY_PATH
+echo $MAVEN_DIST_DIR
+mvn -DdistributionTargetDir=$MAVEN_DIST_DIR -DskipTests clean package %{?offline_build}
+
+%check
```

Comment:
This check section should be running some kind of maven self tests if possible

Reply:
I checked fedora 36 spec file for reference and there is no check section either.